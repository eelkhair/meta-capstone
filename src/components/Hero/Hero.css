.hero {
    background-color: #495e57;
    padding: 2rem 0 1.5rem;
    overflow: visible;
    max-height: 400px;
    position: relative;
    z-index: 0;
    margin-bottom: 6rem;
}

.hero-inner {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
    gap: 2rem;
}

.hero-text {
    max-width: 400px;
    color: white;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
}

.hero-text h1 {
    font-size: 2.8rem;
    color: #f4ce14;
    margin-bottom: 0.5rem;
    text-align: left;
}

.hero-text h2 {
    font-size: 1.4rem;
    color: #ccc;
    margin-bottom: 1rem;
}

.hero-text p {
    margin-bottom: 1rem;
    line-height: 1.6;
    font-size: 1.2rem;
}

.reserve-btn {
    display: inline-block;
    padding: 0.8rem 1.2rem;
    background-color: #f4ce14;
    color: black;
    border: none;
    border-radius: 8px;
    font-weight: bold;
    cursor: pointer;
    transition: background-color 0.3s ease;
    margin: 0;
    font-size: 1.2rem;
    width: 200px;
}

.reserve-btn:hover {
    background-color: #e2ba0f;
}

.hero-image-wrapper {
    flex-shrink: 0;
    transform: translateY(0%);
    z-index: 1;
}

.hero-image-wrapper img {
    width: 260px;
    max-height: 520px;
    object-fit: cover;
    border-radius: 12px;
    box-shadow: 0 12px 24px rgba(0, 0, 0, 0.25);
}

/* Responsive Design */
@media (max-width: 1024px) {
    .hero {
        max-height: unset;
    }

    .hero-inner {
        flex-direction: column;
        align-items: center;
        text-align: center;
    }

    .hero-text {
        align-items: center;
    }

    .hero-image-wrapper {
        transform: none;
        margin-top: 2rem;
        width: 100%;
        max-width: 320px;
        text-align: center;
    }

    .hero-image-wrapper img {
        width: 100%;
        max-width: 100%;
        height: auto;
        display: block;
        margin: 0 auto;
    }

    .hero {
        max-height: unset;
    }

    .hero-inner {
        flex-direction: column;
        align-items: center;
        text-align: center;
    }

    .hero-text {
        align-items: center;
    }

    .hero-image-wrapper {
        transform: translateY(10%);
        margin-top: 2rem;
        max-width: 320px;
    }

    .hero-image-wrapper img {
        width: 100%;
        height: auto;
        margin: 0 auto;
    }
}
