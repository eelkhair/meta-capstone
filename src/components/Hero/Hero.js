import img from '../../assets/restauranfood.jpg';
import './Hero.css';

export default function Hero() {
    return (
        <section className="hero" data-testid="hero">
            <div className="container" data-testid="container">
                <div className="hero-inner" data-testid="hero-inner">
                    <div className="hero-text" data-testid="hero-text">
                        <h1>Little Lemon</h1>
                        <h2>Chicago</h2>
                        <p data-testid="hero-description">
                            We are a family owned Mediterranean restaurant,
                            focused on traditional recipes served with a modern
                            twist.
                        </p>
                        <button
                            className="reserve-btn"
                            data-testid="reserve-button"
                        >
                            Reserve a Table
                        </button>
                    </div>

                    <div
                        className="hero-image-wrapper"
                        data-testid="hero-image-wrapper"
                    >
                        <img
                            src={img}
                            alt="Chef holding appetizers"
                            data-testid="hero-image"
                        />
                    </div>
                </div>
            </div>
        </section>
    );
}
