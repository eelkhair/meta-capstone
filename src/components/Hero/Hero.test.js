import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Hero from './Hero';

jest.mock('../../assets/restauranfood.jpg', () => 'restauranfood.jpg');

describe('Hero Component Tests', () => {
    test('renders the hero section', () => {
        render(<Hero />);
        expect(screen.getByTestId('hero')).toBeInTheDocument();
    });

    test('renders the container div', () => {
        render(<Hero />);
        expect(screen.getByTestId('container')).toBeInTheDocument();
    });

    test('renders the hero-inner wrapper', () => {
        render(<Hero />);
        expect(screen.getByTestId('hero-inner')).toBeInTheDocument();
    });

    test('renders the hero-text block', () => {
        render(<Hero />);
        expect(screen.getByTestId('hero-text')).toBeInTheDocument();
    });

    test('renders the hero image wrapper', () => {
        render(<Hero />);
        expect(screen.getByTestId('hero-image-wrapper')).toBeInTheDocument();
    });

    test('displays correct main heading text', () => {
        render(<Hero />);
        expect(
            screen.getByRole('heading', { name: /little lemon/i })
        ).toBeInTheDocument();
    });

    test('displays correct subheading text', () => {
        render(<Hero />);
        expect(
            screen.getByRole('heading', { name: /chicago/i })
        ).toBeInTheDocument();
    });

    test('displays description paragraph', () => {
        render(<Hero />);
        expect(screen.getByTestId('hero-description')).toHaveTextContent(
            /family owned mediterranean restaurant/i
        );
    });

    test('renders the reserve button', () => {
        render(<Hero />);
        expect(screen.getByTestId('reserve-button')).toBeInTheDocument();
    });

    test('reserve button contains correct text', () => {
        render(<Hero />);
        expect(
            screen.getByRole('button', { name: /reserve a table/i })
        ).toBeInTheDocument();
    });

    test('renders the hero image', () => {
        render(<Hero />);
        expect(screen.getByTestId('hero-image')).toBeInTheDocument();
    });

    test('hero image has correct alt text', () => {
        render(<Hero />);
        expect(
            screen.getByAltText(/chef holding appetizers/i)
        ).toBeInTheDocument();
    });

    test('hero image src matches mock', () => {
        render(<Hero />);
        expect(screen.getByTestId('hero-image').src).toContain(
            'restauranfood.jpg'
        );
    });

    test('hero section has correct class', () => {
        render(<Hero />);
        expect(screen.getByTestId('hero')).toHaveClass('hero');
    });

    test('reserve button has reserve-btn class', () => {
        render(<Hero />);
        expect(screen.getByTestId('reserve-button')).toHaveClass('reserve-btn');
    });

    test('image wrapper has expected class', () => {
        render(<Hero />);
        expect(screen.getByTestId('hero-image-wrapper')).toHaveClass(
            'hero-image-wrapper'
        );
    });

    test('matches snapshot', () => {
        const { asFragment } = render(<Hero />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('has a single heading level 1', () => {
        render(<Hero />);
        const h1s = screen.getAllByRole('heading', { level: 1 });
        expect(h1s.length).toBe(1);
    });

    test('has a single button on the screen', () => {
        render(<Hero />);
        const buttons = screen.getAllByRole('button');
        expect(buttons.length).toBe(1);
    });
});
