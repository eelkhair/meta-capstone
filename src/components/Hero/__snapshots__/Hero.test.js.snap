// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component Tests matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="hero"
    data-testid="hero"
  >
    <div
      class="container"
      data-testid="container"
    >
      <div
        class="hero-inner"
        data-testid="hero-inner"
      >
        <div
          class="hero-text"
          data-testid="hero-text"
        >
          <h1>
            Little Lemon
          </h1>
          <h2>
            Chicago
          </h2>
          <p
            data-testid="hero-description"
          >
            We are a family owned Mediterranean restaurant, focused on traditional recipes served with a modern twist.
          </p>
          <button
            class="reserve-btn"
            data-testid="reserve-button"
          >
            Reserve a Table
          </button>
        </div>
        <div
          class="hero-image-wrapper"
          data-testid="hero-image-wrapper"
        >
          <img
            alt="Chef holding appetizers"
            data-testid="hero-image"
            src="restauranfood.jpg"
          />
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
